src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/InfrastructureFormula.kt

```diff
@@ -31,8 +31,16 @@ class InfrastructureFormula<out T : VirtualUsers>(
     private val investment: Investment,
     private val jiraFormula: JiraFormula,
     private val virtualUsersFormula: VirtualUsersFormula<T>,
-    private val aws: Aws
+    private val aws: Aws,
+    private var preProvisionedNetwork: Network?
```

Comment:
The nullability here represents optional behavior (provisioning a network or not).
Perhaps we could replace it with bona fide polymorphism. E.g. `Supplier<Network>`. We'd need such [indirection anyway, if we go for builders](https://github.com/wyrzyk/java-api-design-checklist/pull/10).

Reply:
Hmm, we won't need it for builders if the default is `null`, fair enough